### PR TITLE
added in billing_date field to subscription but it doesn't update

### PR DESCRIPTION
**Pull request by ninagong**
I added in billing_date field to subscriptions which takes in user input on click. Seems to be fine when i print out the variable on line 118 of NewGroup.tsx but in the end it gives me undefined and that gets posted to the database instead.

### CODE REVIEW

**Comment by Winxton** [comment]
Where's the frontend code where the date selector is? That may be the part that's not getting passed properly. Make sure to print it out at each step to see: after user input, before sending API, before sending to DB, etc

Another issue is I see that the billing_date field in supabase has the type "text". It'll need to be datetime type.

**Comment by ninagong** [comment]
So the date is showing up correctly before sending it to API, also showing up correctly in database.ts before sending it to database, so it would be a database issue? 
I made it to text format for simplicity for now and on Supabase i set the billing_date format to text temporarily but it seems like that's not even being posted and just shows up as 'null'. NEED HELP

**Comment by allenchen2244** [comment]
> So the date is showing up correctly before sending it to API, also showing up correctly in database.ts before sending it to database, so it would be a database issue? I made it to text format for simplicity for now and on Supabase i set the billing_date format to text temporarily but it seems like that's not even being posted and just shows up as 'null'. NEED HELP

The issue is the variable is not being stored in the DB as intended but showing up as NULL? @Winxton Do have a copy of the db schema somewhere?

**Comment by allenchen2244** [comment]
@ninagong disregarding whatever is on the FE. If you hard code a value in the repository layer that you think should work would it show up in the DB?

**Comment by ninagong** [comment]
adding in billing_date field

File: frontend/src/components/NewGroup.tsx [hunk 1]
```diff
@@ -171,13 +179,48 @@ function NewGroup(props: NewGroupProps) {
...
+
+    const formattedDate = isoString.split("T")[0];
 
+    if (date instanceof Date && !isNaN(date.getTime())) {
+      const year = date.getFullYear();
+      const month = String(date.getMonth() + 1).padStart(2, "0");
+      const day = String(date.getDate()).padStart(2, "0");
+      return `${year}-${month}-${day}`;
+    } else {
+      return "";
```
Comment (Winxton) [inline frontend/src/components/NewGroup.tsx]: Would the date ever be invalid here? If there is an unexpected error, then there should be an exception

File: frontend/src/components/NewGroup.tsx [hunk 2]
```diff
@@ -171,13 +179,48 @@ function NewGroup(props: NewGroupProps) {
                   cost: parseInt(e.target.value) || 0,
                 } as Subscription);
               }}
+              mr="2" // Add margin right for spacing
             />
           </InputGroup>
+
+          {/* Billing date input */}
```
Comment (Winxton) [inline frontend/src/components/NewGroup.tsx]: nit: rm comments

**Review by Winxton — APPROVED** [review]

**Pull request merged**